components/script/dom/webgl2renderingcontext.rs

```diff
@@ -0,0 +1,908 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// https://www.khronos.org/registry/webgl/specs/latest/2.0/webgl.idl
+use canvas_traits::webgl::WebGLVersion;
+use dom::bindings::codegen::Bindings::WebGL2RenderingContextBinding;
+use dom::bindings::codegen::Bindings::WebGL2RenderingContextBinding::WebGL2RenderingContextMethods;
+use dom::bindings::codegen::Bindings::WebGLRenderingContextBinding::WebGLContextAttributes;
+use dom::bindings::codegen::Bindings::WebGLRenderingContextBinding::WebGLRenderingContextMethods;
+use dom::bindings::codegen::UnionTypes::ImageDataOrHTMLImageElementOrHTMLCanvasElementOrHTMLVideoElement;
+use dom::bindings::error::Fallible;
+use dom::bindings::nonnull::NonNullJSObjectPtr;
+use dom::bindings::reflector::{reflect_dom_object, Reflector};
+use dom::bindings::root::{Dom, DomRoot, LayoutDom};
+use dom::bindings::str::DOMString;
+use dom::htmlcanvaselement::HTMLCanvasElement;
+use dom::htmliframeelement::HTMLIFrameElement;
+use dom::webglactiveinfo::WebGLActiveInfo;
+use dom::webglbuffer::WebGLBuffer;
+use dom::webglframebuffer::WebGLFramebuffer;
+use dom::webglprogram::WebGLProgram;
+use dom::webglrenderbuffer::WebGLRenderbuffer;
+use dom::webglrenderingcontext::{LayoutCanvasWebGLRenderingContextHelpers, WebGLRenderingContext};
+use dom::webglshader::WebGLShader;
+use dom::webglshaderprecisionformat::WebGLShaderPrecisionFormat;
+use dom::webgltexture::WebGLTexture;
+use dom::webgluniformlocation::WebGLUniformLocation;
+use dom::window::Window;
+use dom_struct::dom_struct;
+use euclid::Size2D;
+use js::jsapi::{JSContext, JSObject};
+use js::jsval::JSVal;
+use offscreen_gl_context::GLContextAttributes;
+use script_layout_interface::HTMLCanvasDataSource;
+
+#[dom_struct]
+pub struct WebGL2RenderingContext {
+    reflector_: Reflector,
+    base: Dom<WebGLRenderingContext>,
+}
+
+impl WebGL2RenderingContext {
+    fn new_inherited(window: &Window,
```

Comment:
nit: For new code, I think it's preferable to use block indentation for arguments and such, like:

```rust
fn new_inherited(
    window: &Window,
    canvas: &HTMLCanvasElement,
    size: Size2D<i32>,
    attrs: GLContextAttributes,
) -> Option<Self> {
```

This applies to other parts of the patch, feel free to reformat the signatures you're touching too if you want.